### PR TITLE
Added support for configuring ConnectionConfigurationFactory parameter

### PR DESCRIPTION
**Pull request by dsenic**
Added parameter which enables to specify custom connection configuration factory class.

I had an issue that default ping timeout for ealsticsearch connection was not good enough for my case.

### CODE REVIEW

File: src/NLog.Targets.ElasticSearch/ElasticSearchTarget.cs [hunk 1]
```diff
@@ -51,7 +54,27 @@ protected override void InitializeTarget()
             var uri = GetConnectionString(ConnectionStringName) ?? Uri;
             var nodes = uri.Split(',').Select(url => new Uri(url));
             var connectionPool = new StaticConnectionPool(nodes);
-            var config = new ConnectionConfiguration(connectionPool);
+
+            IElasticSearchConnectionConfigurationFactory connectionConfigurationFactory = null;
+
+            if ( !string.IsNullOrEmpty( ConnectionConfigurationFactory ) )
+            {
+                var type = Assembly.GetEntryAssembly().GetType( ConnectionConfigurationFactory );
```
Comment (markmcdowell) [inline src/NLog.Targets.ElasticSearch/ElasticSearchTarget.cs]: Move all this code to an extensions

File: src/NLog.Targets.ElasticSearch/ElasticSearchTarget.cs [hunk 2]
```diff
@@ -51,7 +54,27 @@ protected override void InitializeTarget()
             var uri = GetConnectionString(ConnectionStringName) ?? Uri;
             var nodes = uri.Split(',').Select(url => new Uri(url));
             var connectionPool = new StaticConnectionPool(nodes);
-            var config = new ConnectionConfiguration(connectionPool);
+
+            IElasticSearchConnectionConfigurationFactory connectionConfigurationFactory = null;
```
Comment (markmcdowell) [inline src/NLog.Targets.ElasticSearch/ElasticSearchTarget.cs]: Add a property to the class to set this directly

**Comment by markmcdowell** [comment]
Hi dseni, thanks for PR. I'll accept if you make a couple of changes.

**Comment by dsenic** [comment]
Hi,

thanks for pointing that out. I made some changes following your suggestions. Please take a look if it makes sense now.

**Pull request closed without merging**